### PR TITLE
New style mean_squared_error

### PR DESCRIPTION
**Pull request by delta2323**
This PR is a part of #3147 and apply the new-style API to `mean_squared_error`.

### CODE REVIEW

File: chainer/functions/loss/mean_squared_error.py [hunk 1]
```diff
@@ -17,21 +18,30 @@ def check_type_forward(self, in_types):
...
         return diff.dot(diff) / diff.dtype.type(diff.size),
 
-    def backward(self, inputs, gy):
-        coeff = gy[0] * gy[0].dtype.type(2. / self.diff.size)
-        gx0 = coeff * self.diff
-        return gx0, -gx0
+    def backward(self, indexes, gy):
+        x0, x1 = self.get_retained_inputs()
+        ret = []
+        diff = x0 - x1
```
Comment (unnonouno) [inline chainer/functions/loss/mean_squared_error.py]: `diff` is calculated here. You don't need to store `self.diff` in forward_xxx.

File: chainer/functions/loss/mean_squared_error.py [hunk 2]
```diff
@@ -1,10 +1,11 @@
 import numpy
 
-from chainer import function
+from chainer import function_node
+from chainer.functions.array import broadcast
```
Comment (unnonouno) [inline chainer/functions/loss/mean_squared_error.py]: Use `chainer.functions.broadcast_to`

**Comment by delta2323** [comment]
Thank you for your review. Updated.

**Comment by unnonouno** [comment]
jenkins test this please

**Comment by unnonouno** [comment]
jenkins test this please

**Comment by unnonouno** [comment]
LGTM

**Pull request merged**